p256/src/arithmetic/affine.rs

```diff
@@ -58,6 +59,21 @@ impl AffinePoint {
     pub fn is_identity(&self) -> Choice {
         self.infinity
     }
+
+    /// Recover compact point
+    pub fn recover(&self, x: &FieldElement) -> CtOption<FieldElement> {
+        let mut x_cube = x.mul(x).mul(x);
+        let a_times_x = x.mul(&CURVE_EQUATION_A);
+        x_cube.add(&a_times_x).add(&CURVE_EQUATION_B).sqrt()
+            .map(|y| {
```

Comment:
is this a clamping operation?

Reply:
Sort of? We're trying to evaluate `y` from given `x`. The way to do that is over [here](https://github.com/helium/ecc_compact/blob/master/c_src/ecc_compact.c#L122)

Reply:
The problem right now is that we don't have a way of comparing `FieldElements` as the traits `ConstantTimeGreater` and `ConstantTimeLess` aren't implemented for FieldElement